llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -984,9 +985,15 @@ struct DSEState {
       return isMaskedStoreOverwrite(KillingI, DeadI, BatchAA);
     }
 
-    const uint64_t KillingSize = KillingLocSize.getValue();
-    const uint64_t DeadSize = DeadLoc.Size.getValue();
+    const TypeSize KillingSize = KillingLocSize.getValue();
+    const TypeSize DeadSize = DeadLoc.Size.getValue();
+    const bool AnyScalable =
+        DeadSize.isScalable() || KillingLocSize.isScalable();
 
+    // TODO: Remove AnyScalable constraint once alias analysis fully support
+    // scalable quantities
```

Comment:
For the purposes of just this code, is AA support needed, rather than adjusting the size comparisons to use the proper helpers. E.g. in the MustAlias case (which can be determined without extra AA support), we can replace `KillingSize >= DeadSize` with `TypeSize::isKnownGE(KillingSize, DeadSize)` or something?

Reply:
Can we skip only this code path for scalable vectors then?